tests/unit/s2n_random_test.c

```diff
@@ -793,24 +793,25 @@ static int s2n_random_rand_bytes_after_cleanup_cb(struct random_test_case *test_
 
 static int s2n_random_rand_bytes_before_init(struct random_test_case *test_case)
 {
-#if S2N_LIBCRYPTO_SUPPORTS_CUSTOM_RAND
-    /* Calling RAND_bytes will set a global random method */
-    unsigned char rndbytes[16] = { 0 };
-    EXPECT_EQUAL(RAND_bytes(rndbytes, sizeof(rndbytes)), 1);
-    const RAND_METHOD *rand_method = RAND_get_rand_method();
-    EXPECT_NOT_NULL(rand_method);
-    EXPECT_NOT_EQUAL(rand_method->bytes, s2n_openssl_compat_rand);
-
-    EXPECT_SUCCESS(s2n_init());
+#if S2N_LIBCRYPTO_SUPPORTS_ENGINE
```

Comment:
Does compiling this code trigger an error if S2N_LIBCRYPTO_SUPPORTS_ENGINE isn't set? What method are you trying to gate?

We try to limit the use of conditional compilation to ideally one source file. Sometimes it leaks, but that's the goal. You can limit the leakage by not defining any s2n methods that only exist when S2N_LIBCRYPTO_SUPPORTS_ENGINE is defined: always define stub versions.

Reply:
Made it more specific. We need to gate due to signature difference in `RAND_METHOD`. Example [failure](https://us-west-2.codebuild.aws.amazon.com/project/eyJlbmNyeXB0ZWREYXRhIjoidUpNb2QzNjZzQ3lrWm1UQ2xzdFZMN1ZGYWE5dVlSQ1ZKd1BnZzZ6aXN1dlZXTmFuZWdvNnR0SitPaFlIcUdjaDZ6RU9KemY2Mi9ybXVBajF0RXo4NmZ1Rnl4dkZrT241UUY4MTVQMkRuNTQ9IiwiaXZQYXJhbWV0ZXJTcGVjIjoiemFRUDgvZUMreVhER0RKaiIsIm1hdGVyaWFsU2V0U2VyaWFsIjoxfQ%3D%3D/build/258d5a62-c224-4486-91a4-d8906dc33715)